File: Data Structure/BubbleSort.java
import java.lang.*;
import java.util.*;


class Bubble
{
	public void Sort(int arr[])
	{
		int pass=0,j=0,no=0,temp=0;
		boolean flag=true;

		for(pass=0;pass<arr.length && flag==true;pass++)
		{
			no=pass+1;
			flag=false;

			for(j=0;j<arr.length-pass-1;j++)
			{
				if(arr[j]>arr[j+1])
				{
					flag=true;
					temp=arr[j];
					arr[j]=arr[j+1];
					arr[j+1]=temp;
				}
			}
		}
	}
}

class BubbleSort
{
	public static void main(String args[])
	{
		Scanner obj=new Scanner(System.in);

		System.out.println("Enter size of array :");
		int no=obj.nextInt();

		int arr[]=new int[no];

		System.out.println("Enter elements of array :");
		for(int i=0;i<arr.length;i++)
		{
			arr[i]=obj.nextInt();
		}

		System.out.println("Entered elements of array are:");
		for(int j=0;j<arr.length;j++)
		{
			System.out.print(arr[j]+"\t");
		}

		Bubble obj1=new Bubble();

		obj1.Sort(arr);

		System.out.println("\nElements of array after Sorting are:");
		for(int k=0;k<arr.length;k++)
		{
			System.out.print(arr[k]+"\t");
		}

	}
}

File: Data Structure/BubbleSort.c
#include<stdio.h>
#include<stdlib.h>

void BubbleSort(int *arr,int no)
{
	for(int i=0;i<no;i++)
	{
		for(int j=0;j<no-i-1;j++)
		{
			if(arr[j]>arr[j+1])
			{
				int temp=arr[j];
				arr[j]=arr[j+1];
				arr[j+1]=temp;
			}
		}
	}
}

int main()
{
	int no=0;
	int *arr=NULL;

	printf("Enter number of elements in array are :");
	scanf("%d",&no);

	arr=(int*)malloc(no * sizeof(int));

	printf("Enter elements of array :\n");
	for(int i=0;i<no;i++)
	{
		scanf("%d",&arr[i]);
	}

	printf("Entered elements of array are:\n");
	for(int j=0;j<no;j++)
	{
		printf("%d\t",arr[j]);
	}

	printf("\nArray after BubbleSort is :\n");
	BubbleSort(arr,no);

	for(int k=0;k<no;k++)
	{
		printf("%d\t",arr[k]);
	}
	
	return 0;
}